controllers/api/base_api_controller.js

```diff
@@ -15,56 +15,135 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-module.exports = function(pb) {
+module.exports = function BaseApiControllerModule(pb) {
+
+  // PB dependencies
+  const util = pb.util;
+  const BaseObjectService = pb.BaseObjectService;
+
+  /**
+   *
+   * @class BaseApiController
+   * @constructor
+   */
+  class BaseApiController extends pb.BaseController {
+
+    constructor() {
+      super();
+      /**
+       * Processes the query string to develop the where clause for the query request
+       * @method processWhere
+       * @param {Object} q The hash of all query parameters from the request
+       * @return {Object}
+       */
+      this.processWhere = function processWhere() {
```

Comment:
There are many instance functions that not use the instance (this), I put them inside the constructor because this rule http://eslint.org/docs/rules/class-methods-use-this, I think that this functions should be statics
I didn't want to do a deeper refactoring because I still don't understand all pencilblue ecosystem